aws.go

```diff
@@ -39,13 +40,34 @@ type AwsPlugin struct {
 }
 
 func (plugin AwsPlugin) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "PUT":
+		plugin.put(rw, req)
+	case "GET":
+		plugin.get(rw, req)
+	default:
+		http.Error(rw, fmt.Sprintf("Method %s not implemented", req.Method), http.StatusNotImplemented)
+	}
+	plugin.next.ServeHTTP(rw, req)
+}
+
+func (plugin AwsPlugin) put(rw http.ResponseWriter, req *http.Request) {
 	payload, err := io.ReadAll(req.Body)
 	if err != nil {
 		rw.WriteHeader(http.StatusNotAcceptable)
 		log.Error(fmt.Sprintf("Reading body failed: %s", err.Error()))
 		return
 	}
 	resp, err := plugin.service.Put(req.URL.Path[1:], payload, req.Header.Get("Content-Type"), rw)
+	handleResponse(resp, err, rw)
+}
+
+func (plugin *AwsPlugin) get(rw http.ResponseWriter, req *http.Request) {
+	resp, err := plugin.service.Get(req.URL.Path[1:], rw)
+	handleResponse(resp, err, rw)
+}
+
+func handleResponse(resp []byte, err error, rw http.ResponseWriter) {
```

Comment:
Can be confusing to use the name `err` as an argument and internal variable. I would rename the argument, maybe to `requestErr`